### PR TITLE
fix: [PL-30081]: Removed redundant subsequence indexes related to delegate

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
…egate

## Describe your changes
As a part of cleanup of redundant indexes to improve mongo, we are removing all the indexes which are a subsequence of other indexes. This pr includes those redundant indexes cleanup which are related to delegate. 
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)

### CODE REVIEW

**Review by arpit3107 — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 3 file(s) to review. 1 of 2 approvals obtained.

File: 420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java [hunk 1]
```diff
@@ -68,11 +68,6 @@ public static List<MongoIndex> mongoIndexes() {
                  .field(PerpetualTaskRecordKeys.state)
                  .field(PerpetualTaskRecordKeys.rebalanceIteration)
                  .build())
-        .add(CompoundMongoIndex.builder()
-                 .name("assignIterator_1")
-                 .field(PerpetualTaskRecordKeys.state)
-                 .field(PerpetualTaskRecordKeys.assignIteration)
```
Comment (arpit3107) [inline 420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java:74]: need to remove this index name from some other files also. I think UTs will tell you that

**Comment by jennyjames21** [comment]
*[`420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java` line 79 at r1](https://reviewable.io/reviews/harness/harness-core/40769#-NI0tsjAAHZMmnwxUEJA:-NI0tsjAAHZMmnwxUEJB:b-cwu93e) ([raw file](https://github.com/harness/harness-core/blob/7781e543991d16029370cfb77542dec10f09f074/420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java#L79)):*
> ```Java
>                  .name("state_assignIteration_assignAfterMs")
>                  .field(PerpetualTaskRecordKeys.state)
>                  .field(PerpetualTaskRecordKeys.assignIteration)
> ```

how did you conclude this order? for previously created state_assignIteration_assignAfterMs

IMO we should remove only after new indexes run in production fine?

**Comment by raghuAtWings** [comment]
*[`420-delegate-service/src/main/java/io/harness/delegate/beans/Delegate.java` line 72 at r1](https://reviewable.io/reviews/harness/harness-core/40769#-NI0zGDADGgMn-G91gFx:-NI0zGDADGgMn-G91gFy:b-vuqirc) ([raw file](https://github.com/harness/harness-core/blob/7781e543991d16029370cfb77542dec10f09f074/420-delegate-service/src/main/java/io/harness/delegate/beans/Delegate.java#L72)):*
> ```Java
>   @SchemaIgnore @FdIndex private long createdAt;
>   // Will be used by ECS delegate, when hostName is mentioned in TaskSpec.
>   @NotEmpty @FdIndex private String accountId;
> ```

why is this removed?

**Comment by raghuAtWings** [comment]
*[`420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java` line 79 at r1](https://reviewable.io/reviews/harness/harness-core/40769#-NI0tsjAAHZMmnwxUEJA:-NI0zY_t54L190G4VTg4:bczlhtm) ([raw file](https://github.com/harness/harness-core/blob/7781e543991d16029370cfb77542dec10f09f074/420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java#L79)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

how did you conclude this order? for previously created state_assignIteration_assignAfterMs

IMO we should remove only after new indexes run in production fine?
</blockquote></details>

i think the above is removed since `state_assignIteration_assignAfterMs` will already have the index that is being deleted.

**Review by raghuAtWings — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 1. unresolved discussions.

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 1. unresolved discussions. 

___
*[`420-delegate-service/src/main/java/io/harness/delegate/beans/Delegate.java` line 72 at r1](https://reviewable.io/reviews/harness/harness-core/40769#-NI0zGDADGgMn-G91gFx:-NI110Ok9fdizV2Ckhea:bq4wlue) ([raw file](https://github.com/harness/harness-core/blob/7781e543991d16029370cfb77542dec10f09f074/420-delegate-service/src/main/java/io/harness/delegate/beans/Delegate.java#L72)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

why is this removed?
</blockquote></details>

There is a compound index which covers accountId
___
*[`420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java` line 79 at r1](https://reviewable.io/reviews/harness/harness-core/40769#-NI0tsjAAHZMmnwxUEJA:-NI11FEF6AqZ2IWcN6H0:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7781e543991d16029370cfb77542dec10f09f074/420-delegate-service/src/main/java/io/harness/perpetualtask/internal/PerpetualTaskRecord.java#L79)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

i think the above is removed since `state_assignIteration_assignAfterMs` will already have the index that is being deleted. 
</blockquote></details>

Done.

File: 930-delegate-tasks/src/main/java/software/wings/service/impl/newrelic/NewRelicMetricDataRecord.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ public static List<MongoIndex> mongoIndexes() {
 
   private String serviceId;
 
-  @FdIndex private String cvConfigId;
```
Comment (abhinav-harness) [inline 930-delegate-tasks/src/main/java/software/wings/service/impl/newrelic/NewRelicMetricDataRecord.java:135]: This shouldnt be removed

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769#-:-NIEJzsQBv93pDS_i42K:bnfp4nl)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by kshitij-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 1. unresolved discussions.

**Review by jennyjames21 — COMMENTED** [review]
> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: 1. unresolved discussions.

**Review by jennyjames21 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40769)* status: :shipit: complete! ✓

**Pull request merged**